src/algebra/monoid_algebra.lean

```diff
@@ -193,33 +230,13 @@ lemma single_one_mul_apply (f : monoid_algebra k G) (r : k) (x : G) :
   (single 1 r * f) x = r * f x :=
 f.single_mul_apply_aux $ λ a, by rw [one_mul]
 
-end
-
-instance [comm_semiring k] [comm_monoid G] : comm_semiring (monoid_algebra k G) :=
-{ mul_comm := assume f g,
-  begin
-    simp only [mul_def, finsupp.sum, mul_comm],
-    rw [finset.sum_comm],
-    simp only [mul_comm]
-  end,
-  .. monoid_algebra.semiring }
-
-instance [ring k] : has_neg (monoid_algebra k G) :=
-by apply_instance
```

Comment:
@urkud: This instance was removed as it relied on local attribute [reducible] monoid_algebra...